Edit several articles (#184)
microsoft/VSExtensibility: docs/new-extensibility-model/getting-started/oop-extensibility-model-overview.md

---
title: OOP Extensibility Model reference
description: A reference for OOP Extensibility Model
date: 2022-2-6
---

# Visual Studio.Extensibility overview

> While the existing model in the VS SDK loads extensions in-process, the new extensibility model, VisualStudio.Extensibility, brings most Visual Studio extensions out-of-process. This out-of-proc model gives you the opportunity to create more reliable, secure, and easier-to-write extensions while still providing the in-depth functionality the old model provides. The following documentation describes:

Reviewer: ```suggestion
While extensions have typically been loaded into the Visual Studio process, VisualStudio.Extensibility, brings most Visual Studio extensions out-of-process. This out-of-proc model gives you the opportunity to create more reliable, secure, and easier-to-write extensions. The following documentation describes:
```

* The general architecture of the new extensibility model
* How to take advantage of the new extensibility model's APIs
* How to compile and F5 debug an extension with the new model 
* Resources and code samples to get started writing an extension with the new model

## About the VisualStudio.Extensibility model

> The new VisualStudio.Extensibility model aims to address many of the problems both developers experience when using and writing extensions in Visual Studio using the old model. These issues include:

Reviewer: Same comment about "both" and flipping these to be positives

Reviewer: Oh, maybe these can't be flipped because the positives are below. In that case, probably eliminate "tribal knowledge" and reword the last one to "No hot-reloading of newly installed extensions"

* Extension-caused crashes and hangs of Visual Studio and other installed extensions

> * Inconsistent hard-to-find docs and APIs, tribal knowledge requirements, and overwhelming architecture

Author: ```suggestion
* Inconsistent hard-to-find docs and APIs, requirement for specialized expertise, and overwhelming architecture
```

* Having to restart Visual Studio when installing extensions

> Writing extensions using the VisualStudio.Extensibility provides the following benefits:

Reviewer: ```suggestion
Writing extensions using VisualStudio.Extensibility provides the following benefits:
```
